### ISSUE TITLE
iframe style tags are stripped by Vite 2.8.5

### ISSUE THREAD

**Issue by justrhysism** [eirslett/storybook-builder-vite#266]
When Vite 2.8.5 executes `transformIndexHtml` and internally runs "devHtmlHook", all the `<style>` are stripped from the iframe `<head>` resulting in Storybook's template code being exposed:

![image](https://user-images.githubusercontent.com/1326759/156121496-3e764360-be15-4ac5-94dd-da93bf91e9e7.png)

Have confirmed that this issue is **new for 2.8.5** (downgrading to 2.8.4 doesn't exhibit this behaviour).

**Comment by IanVS** [comment]
Thanks, I'm working with the vite folks to figure out what's going wrong.  https://github.com/vitejs/vite/issues/7124

**Comment by Dschungelabenteuer** [comment]
https://github.com/eirslett/storybook-builder-vite/blob/main/packages/storybook-builder-vite/index.ts#L20

Now that Vite uses `?html-proxy` to handle HMR for style tags in HTML files ([vitejs/vite#7052](https://github.com/vitejs/vite/pull/7052) as you pointed out), it seems to match the regex here and therefore serve the whole `iframe.html` again instead of the vite-generated asset.

I've tried to tweak the condition into something silly like this:

```ts
if (req.url.match(/\?html-proxy/) || !req.url.match(/^\/iframe.html($|\?)/)) {
    next();
    return;
  }
```

This seems to work fine in my case but this requires additional checks with Storybook's query parameters.

**Comment by justrhysism** [comment]
Have confirmed this issue is fixed for **Vite 2.8.6**.

Will close as there are other issues tracking the underlying problem. Thanks everyone!
